frontend/src/app/main/ui/workspace/sidebar/options/rows/color_row.cljs

```diff
@@ -45,18 +45,18 @@
   (if (= v :multiple) nil v))
 
 (mf/defc color-row
-  [{:keys [index color disable-gradient disable-opacity disable-image disable-picker on-change
-           on-reorder on-detach on-open on-close on-remove
+  [{:keys [index color disable-gradient disable-opacity disable-image disable-picker hidden
+           on-change on-reorder on-detach on-open on-close on-remove
            disable-drag on-focus on-blur select-only select-on-focus]}]
   (let [shared-libs      (mf/deref refs/libraries)
         hover-detach     (mf/use-state false)
         on-change        (h/use-ref-callback on-change)
 
-        src-colors       (dm/get-in shared-libs [(:ref-file color) :data :colors])
-        color-name       (dm/get-in src-colors [(:ref-id color) :name])
+        src-colors       (dm/get-in shared-libs [(or (:ref-file color) (:file-id color)) :data :colors])
+        color-name       (dm/get-in src-colors [(or (:ref-id color) (:id color)) :name])
```

Comment:
i think this or's should be extracted to separated bindings on the let, this concrete or is repeated some lines below